### PR TITLE
Refactoring Administrator, Home and SignIn pages

### PR DESCRIPTION
**Pull request by oleksandranikitiuk**

### CODE REVIEW

File: src/test/java/pages/HomePage.java [hunk 1]
```diff
@@ -4,33 +4,24 @@
...
+    public HomePage(WebDriver driver)   {
+        this.driver = driver;
+        PageFactory.initElements(this.driver, this);
     }
-    public void clickRestaurantsListBtn() {
-        restaurantsListBtn.click();
+
+    public HeaderGeneralPageComponent getHeaderGeneralPageComponent() {
+        return new HeaderGeneralPageComponent(driver);
     }
```
Comment (gkyslenko) [inline src/test/java/pages/HomePage.java:23]: We can add initialization to the constructor

File: src/test/java/pages/SignInPage.java [hunk 2]
```diff
@@ -18,9 +19,14 @@ public SignInPage(WebDriver driver) {
...
-    @FindBy(xpath = "//*[@id=\"root\"]/header/div/a")
-    private WebElement homeBtn;
+    public SignInPage(WebDriver driver) {
+        this.driver = driver;
+        PageFactory.initElements(this.driver, this);
+    }
 
+    public HeaderGeneralPageComponent getHeaderGeneralPageComponent() {
+        return new HeaderGeneralPageComponent(driver);
+    }
```
Comment (gkyslenko) [inline src/test/java/pages/SignInPage.java:30]: We can add initialization to the constructor

File: src/test/java/pages/administrator/AdministratorPage.java [hunk 3]
```diff
@@ -17,6 +18,10 @@ public AdministratorPage(WebDriver driver) {
         PageFactory.initElements(this.driver, this);
     }
 
+    public HeaderGeneralPageComponent getHeaderGeneralPageComponent() {
+        return new HeaderGeneralPageComponent(driver);
+    }
+
```
Comment (gkyslenko) [inline src/test/java/pages/administrator/AdministratorPage.java:24]: We can add initialization to the constructor

File: src/test/java/pages/HomePage.java (hunk 1 above)
Comment (gkyslenko) [reply src/test/java/pages/HomePage.java:23]: Also don't sure if we need this method at all

Comment (oleksandranikitiuk) [reply src/test/java/pages/HomePage.java:23]: I'm not sure we have to initialize it in the constructor. I think there is no reason to use eager initialization.

**Review by derkachOksana — DISMISSED** [review]

**Review by dmatsev — APPROVED** [review]

**Review by gkyslenko — APPROVED** [review]

**Pull request merged**